src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,247 @@
+    return {
+        "eventId": "Event Id",
+        "eventTimestatmp": "Timestamp",
+        "filePath": "Path",
+        "fileName": "File",
+        "exposure": "Exposure",
+    }
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def cases(state):
+    """For managing cases and events associated with cases."""
+    pass
+
+
+@cases.command()
+@click.argument("name")
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@sdk_options()
+def create(state, name, subject, assignee, description, notes):
+    """Create a new case."""
+    state.sdk.cases.create(
+        name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+    )
+
+
+@cases.command()
+@case_number_arg
+@name_option
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@status_option
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+@cases.command("list")
+@click.option(
+    "--name", help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--subject", help="Filter by user UID of the subject of a case.")
+@click.option("--assignee", help="Filter by user UID of assignee.")
+@click.option("--begin-create-time", **BEGIN_DATE_DICT)
+@click.option("--end-create-time", **END_DATE_DICT)
+@click.option("--begin-update-time", **BEGIN_DATE_DICT)
+@click.option("--end-update-time", **END_DATE_DICT)
+@click.option("--status", help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    begin_create_time,
+    end_create_time,
+    begin_update_time,
+    end_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=begin_create_time,
+        max_create_time=end_create_time,
+        min_update_time=begin_update_time,
+        max_update_time=end_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _get_cases_header())
+    cases = [case for page in pages for case in page["cases"]]
+    if cases:
+        formatter.echo_formatted_list(cases)
+    else:
+        click.echo("No cases found.")
+
+
+def _get_file_events(sdk, case_number):
+    response = sdk.cases.file_events.get_all(case_number)
+    if not response["events"]:
+        return None
+    return json.loads(response.text)
+
+
+def _display_file_events(events):
+    if events:
+        click.echo("\nFile Events:\n")
+        click.echo(pformat(events))
+    else:
+        click.echo("No events found.")
```

Comment:
Could you include a newline here as well, like

`click.echo("\nNo events found.")`

Reply:
wondering whether we should use `click.echo()` over specifying "\n", to make it os independent.